sources/packages/backend/libs/services/src/student/student.service.ts

```diff
@@ -0,0 +1,78 @@
+import { Injectable } from "@nestjs/common";
+import {
+  RecordDataModelService,
+  SINValidation,
+  Student,
+  User,
+} from "@sims/sims-db";
+import { DataSource } from "typeorm";
+import { LoggerService, InjectLogger } from "@sims/utilities/logger";
+
+@Injectable()
+export class StudentService extends RecordDataModelService<Student> {
+  constructor(private readonly dataSource: DataSource) {
+    super(dataSource.getRepository(Student));
+    this.logger.log("[Created]");
+  }
+
+  /**
+   * Gets all the students that have the SIN validation pending.
+   * @returns Students pending SIN validation.
+   */
+  async getStudentsPendingSinValidation(): Promise<Student[]> {
+    return this.repo
+      .createQueryBuilder("student")
+      .select([
+        "student.id",
+        "student.birthDate",
+        "student.gender",
+        "user.firstName",
+        "user.lastName",
+        "user.id",
+        "sinValidation.id",
+        "sinValidation.sin",
+      ])
+      .innerJoin("student.user", "user")
+      .innerJoin("student.sinValidation", "sinValidation")
+      .where("sinValidation.isValidSIN is null")
+      .andWhere("sinValidation.dateSent is null")
+      .getMany();
+  }
+  async getStudentsAppliedForPD(): Promise<Student[]> {
```

Comment:
Please add a comment for this method.

Reply:
these are old files and function. I am just moving it to `libs/`